src/Microsoft.AspNet.Diagnostics.Elm/RequestIdentifier.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNet.Http;
+using Microsoft.AspNet.Http.Features;
+
+namespace Microsoft.AspNet.Diagnostics.Elm
+{
+    internal class RequestIdentifier : IDisposable
+    {
+        private readonly bool _addedFeature;
+        private readonly HttpContext _context;
+
+        private RequestIdentifier(HttpContext context)
+        {
+            _context = context;
+            var feature = context.GetFeature<IHttpRequestIdentifierFeature>();
+            if (feature == null || (feature != null && string.IsNullOrEmpty(feature.TraceIdentifier)))
```

Comment:
If the feature exists but the TraceIdentifier doesn't, just set the TraceIdentifier and not _addedFeature.